Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: a144fb3b8158c7ca0bb5fec4a4b95004f3bea9db
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12241/HPCC-Platform/system/jlib/jmutex.hpp:999:5: error: use of deleted function \xe2\x80\x98std::atomic<_Tp*>::atomic(const std::atomic<_Tp*>&) [with _Tp = OsDiskInfo]\xe2\x80\x99

[Build error:](https://gist.github.com/905b7255889336f6f2d92a5541210ef0)
Build log [Build log](https://gist.github.com/HPCCSmoketest/905b7255889336f6f2d92a5541210ef0#file-result-19-02-28-13-02-18-log)

--- a/system/jlib/jmutex.hpp
+++ b/system/jlib/jmutex.hpp
@@ -999,7 +999,7 @@ public:
     template <typename FUNC> X * query(FUNC factory) { return querySingleton(singleton, cs, factory); }
     X * queryExisting() const { return singleton.load(std::memory_order_acquire); }
 private:
-    std::atomic<X *> singleton{nullptr};
+    std::atomic<X *> singleton = {nullptr};
     CriticalSection cs;
 };
 

